engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanGetter.java

```diff
@@ -47,6 +47,13 @@ public String getPropertyName() {
 		return propertyName;
 	}
 
+	@Override
+	public String getResolvedPropertyName() {
+		// or throw unsupported operation?
+		// or resolve the name
```

Comment:
It's still a property, right? you could have annotated getters that would be converted to json properties. If we follow the approach of resolving the name based on a `Property` this case would get covered as well.